File: code/ch09/9.1.2.2.container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int area = 0;
        int left = 0, right = height.size() - 1;
        
        while(left < right) {
            area = (right - left) * min(height[right], height[left]);
            res  = max(res, area);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        
        return res;
    }
};

File: code/ch09/9.1.2.2.container_with_most_water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, res = 0, area = 0;

        while (left < right) {
            area = Math.min(height[left], height[right]) * (right - left);
            res = Math.max(res, area);
            if (height[left] < height[right])
                left++;
            else
                right++;
        }
        return res;
    }
}